service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -0,0 +1,305 @@
+     */
+    protected final void reply(final Message<JsonObject> request, final TenantResult result) {
+        final JsonObject body = request.body();
+        final String tenantId = body.getString(RequestResponseApiConstants.FIELD_TENANT_ID);
+        request.reply(TenantConstants.getServiceReplyAsJson(tenantId, result));
+    }
+
+    private JsonObject getRequestPayload(final JsonObject request) {
+        final JsonObject payload = request.getJsonObject(TenantConstants.FIELD_PAYLOAD, new JsonObject());
+        addNotPresentFieldsWithDefaultValuesForTenant(payload);
+        return payload;
+    }
+
+    private boolean isValidRequestPayload(final JsonObject request) {
+        final JsonObject payload = request.getJsonObject(TenantConstants.FIELD_PAYLOAD, new JsonObject());
+        final JsonArray adapters = payload.getJsonArray(TenantConstants.FIELD_ADAPTERS);
+        if (adapters != null) {
+            if (adapters.size() == 0) {
+                return false;
+            }
+            for (int i = 0; i < adapters.size(); i++) {
+                final JsonObject adapterDetails = adapters.getJsonObject(i);
+                if (adapterDetails == null) {
+                    return false;
+                }
+                // check for mandatory field adapter type
+                if (adapterDetails.getString(FIELD_ADAPTERS_TYPE) == null) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private void addNotPresentFieldsWithDefaultValuesForTenant(final JsonObject payload) {
+        if(!payload.containsKey(TenantConstants.FIELD_ENABLED)) {
+            log.debug("adding 'enabled' key to payload");
+            payload.put(TenantConstants.FIELD_ENABLED, Boolean.TRUE);
+        }
+
+        final JsonArray adapters = payload.getJsonArray(TenantConstants.FIELD_ADAPTERS);
+        if (adapters != null) {
+            for (int i = 0; i < adapters.size(); i++) {
+                final JsonObject adapterDetails = adapters.getJsonObject(i);
+                addNotPresentFieldsWithDefaultValuesForAdapter(adapterDetails);
+            }
+        }
+    }
+    
+    private void addNotPresentFieldsWithDefaultValuesForAdapter(final JsonObject adapter) {
+        if(!adapter.containsKey(TenantConstants.FIELD_ENABLED)) {
+            log.debug("adding 'enabled' key to payload");
+            adapter.put(TenantConstants.FIELD_ENABLED, Boolean.TRUE);
+        }
+        
+        if(!adapter.containsKey(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED)) {
+            log.debug("adding 'device-authentication-required' key to adapter payload");
+            adapter.put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, Boolean.TRUE);
+        }
+    }
+
+    /**
+     * Wraps a given tenant ID, it's properties data and it's adapter configuration data into a JSON structure suitable
```

Comment:
should read `... its properties...` and `.. its adapter ...`

Reply:
:+1: